### PR TITLE
Validate font file for avatar generation and fix avatar-related unit and integration tests

### LINKED ISSUE #935: Fix tests that are broken by the introduction of the new avatar feature
Fix the following error that resulted from the introduction of the new avatar feature: https://github.com/cortezaproject/corteza/actions/runs/4405206108/jobs/7715797423

### PR DESCRIPTION
**Pull request by KinyaElGrande**
This PR adds validation for the font file used to generate avatar initials and it fixes the tests that were broken by the introduction of the new avatar feature.

### CODE REVIEW

File: server/auth/handlers/handle_profile.go [hunk 1]
```diff
@@ -137,7 +140,8 @@ func (h *AuthHandlers) profileProc(req *request.AuthReq) error {
 		return nil
 	}
 
-	avatarUrl := fmt.Sprintf("%sapi/system/attachment/avatar/%d/original/%s", GetLinks().Base, u.Meta.AvatarID, types.AttachmentKindAvatar)
+	fullpathAPI := "/" + strings.TrimPrefix(options.CleanBase(options.HttpServer().BaseUrl, options.HttpServer().ApiBaseUrl), "/")
```
Comment (petergrlica) [inline server/auth/handlers/handle_profile.go]: duplicate as above, create a private method for this.
also, you can put all the vars in one multi-line Sprintf no?

File: server/system/service/attachment.go [hunk 2]
```diff
@@ -521,3 +522,28 @@ func (svc attachment) processImage(original io.ReadSeeker, att *types.Attachment
...
+
+// processFontsFile validates the file path provided in the AVATAR_INITIALS_FONT_PATH environment variable,
+// It checks if the file exists and has the correct file extension, then reads and returns the file content
+func (svc attachment) processFontsFile() (fontBytes []byte, err error) {
+	aux, err := filepath.Glob(svc.opt.AvatarInitialsFontPath)
+	if err != nil {
+		return nil, err
+	}
+	if aux == nil || len(aux) != 1 {
+		return nil, errors.New("font file not found.please ensure that the correct AVATAR_INITIALS_FONT_PATH is set")
```
Comment (petergrlica) [inline server/system/service/attachment.go]: font file not found, please ....

File: server/system/service/attachment.go [hunk 3]
```diff
@@ -521,3 +522,28 @@ func (svc attachment) processImage(original io.ReadSeeker, att *types.Attachment
...
+	}
+
+	ext := strings.ToLower(filepath.Ext(aux[0]))
+
+	if ext != ".ttf" {
+		return nil, errors.New("invalid font file extension.please provide a truetype font (.ttf) file")
```
Comment (petergrlica) [inline server/system/service/attachment.go]: same as above

**Review by petergrlica — CHANGES_REQUESTED** [review]

File: server/auth/handlers/handle_profile.go (hunk 1 above)
Comment (tjerman) [reply server/auth/handlers/handle_profile.go]: This can also be an anonymous function; pick your poison

**Review by petergrlica — APPROVED** [review]

**Pull request merged**